modules/sdk/gradle-plugins-workspace/src/main/java/com/liferay/gradle/plugins/workspace/configurators/OverridesProjectConfigurator.java

```diff
@@ -0,0 +1,164 @@
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import org.gradle.api.Project;
+import org.gradle.api.file.CopySourceSpec;
+import org.gradle.api.initialization.Settings;
+import org.gradle.api.plugins.ExtensionAware;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.tasks.Copy;
+import org.gradle.jvm.tasks.Jar;
+
+/**
+ * @author David Truong
+ */
+public class OverridesProjectConfigurator extends BaseProjectConfigurator {
+
+	public OverridesProjectConfigurator(Settings settings) {
+		super(settings);
+
+		_defaultRepositoryEnabled = GradleUtil.getProperty(
+			settings,
+			WorkspacePlugin.PROPERTY_PREFIX + NAME +
+				".default.repository.enabled", _DEFAULT_REPOSITORY_ENABLED);
+	}
+
+	@Override
+	public void apply(Project project) {
+		final WorkspaceExtension workspaceExtension = GradleUtil.getExtension(
+			(ExtensionAware)project.getGradle(), WorkspaceExtension.class);
+
+		_applyPlugins(project);
+
+		if (isDefaultRepositoryEnabled()) {
+			GradleUtil.addDefaultRepositories(project);
+		}
+
+		_configureLiferay(project, workspaceExtension);
+
+		Jar jar = (Jar)GradleUtil.getTask(project, JavaPlugin.JAR_TASK_NAME);
+
+		_configureRootTaskDistBundle(jar);
+	}
+
+	@Override
+	public String getName() {
+		return NAME;
+	}
+
+	public boolean isDefaultRepositoryEnabled() {
+		return _defaultRepositoryEnabled;
+	}
+
+	@Override
+	protected Iterable<File> doGetProjectDirs(File rootDir) throws Exception {
+		final Set<File> projectDirs = new HashSet<>();
+
+		Files.walkFileTree(
+			rootDir.toPath(),
+			new SimpleFileVisitor<Path>() {
+
+				@Override
+				public FileVisitResult preVisitDirectory(
+						Path dirPath, BasicFileAttributes basicFileAttributes)
+					throws IOException {
+
+					if (Files.isDirectory(dirPath.resolve("src"))) {
+						projectDirs.add(dirPath.toFile());
+
+						return FileVisitResult.SKIP_SUBTREE;
+					}
+
+					return FileVisitResult.CONTINUE;
+				}
+
+			});
+
+		return projectDirs;
+	}
+
+	protected static final String NAME = "overrides";
+
+	private void _applyPlugins(Project project) {
+		GradleUtil.applyPlugin(project, ModuleExtPlugin.class);
+	}
+
+	private void _configureLiferay(
+		Project project, final WorkspaceExtension workspaceExtension) {
+
+		final LiferayExtension liferayExtension = GradleUtil.getExtension(
+			project, LiferayExtension.class);
+
+		liferayExtension.setAppServerParentDir(workspaceExtension.getHomeDir());
+
+		liferayExtension.setDeployDir(
+			new Callable<File>() {
+
+				@Override
+				public File call() throws Exception {
+					return new File(
+						workspaceExtension.getHomeDir(),
+						"osgi/marketplace/override");
+				}
+
+			});
+	}
+
+	private void _configureRootTaskDistBundle(final Jar jar) {
+		final Project project = jar.getProject();
+
+		Copy copy = (Copy)GradleUtil.getTask(
+			project.getRootProject(),
+			RootProjectConfigurator.DIST_BUNDLE_TASK_NAME);
+
+		copy.into(
+			"osgi/marketplace/override",
```

Comment:
Why is this code both here and in `ModuleExtPlugin`? I don't think you need it here.

Reply:
This is duplicated because I couldn't get it to deploy to the right folder... in workspace it keeps deploying to ..overrides/bundles/ instead of ..rootDir/bundles